File: tests/Generator/resource/visualbasic/complex/GenericPropertyType.vb
Imports System.Text.Json.Serialization

' Represents a generic value which can be replaced with a concrete type
Public Class GenericPropertyType
    Inherits PropertyType
    <JsonPropertyName("name")>
    Public Property Name As Nullable(String)

    <JsonPropertyName("type")>
    Public Property Type As Nullable(String)

End Class

File: tests/Generator/resource/rust/complex/generic_property_type.rs
use serde::{Serialize, Deserialize};
use property_type::PropertyType;

// Represents a generic value which can be replaced with a dynamic type
#[derive(Serialize, Deserialize)]
pub struct GenericPropertyType {
    #[serde(rename = "deprecated")]
    deprecated: Option<bool>,

    #[serde(rename = "description")]
    description: Option<String>,

    #[serde(rename = "nullable")]
    nullable: Option<bool>,

    #[serde(rename = "type")]
    _type: Option<String>,

    #[serde(rename = "name")]
    name: Option<String>,

}

